pywr-schema/src/model.rs

```diff
@@ -424,22 +458,29 @@ impl PywrModel {
             })
             .collect::<Vec<_>>();
 
+        let mut ts_data = nodes_and_ts
+            .iter()
+            .filter_map(|n| n.timeseries.clone())
+            .flatten()
+            .collect::<Vec<_>>();
+
+        let nodes = nodes_and_ts.into_iter().map(|n| n.node).collect::<Vec<_>>();
+
         let edges = v1.edges.into_iter().map(|e| e.into()).collect();
 
-        let parameters = if let Some(v1_parameters) = v1.parameters {
+        let (parameters, param_ts_data) = if let Some(v1_parameters) = v1.parameters {
             let mut unnamed_count: usize = 0;
-            Some(
-                v1_parameters
-                    .into_iter()
-                    .filter_map(|p| match p.try_into_v2_parameter(None, &mut unnamed_count) {
-                        Ok(p) => Some(p),
-                        Err(e) => {
-                            errors.push(e);
-                            None
-                        }
-                    })
-                    .collect::<Vec<_>>(),
-            )
+            let (parameters, param_ts_data) = convert_parameter_v1_to_v2(v1_parameters, &mut unnamed_count, &mut errors);
+            (Some(parameters), Some(param_ts_data))
+        } else {
+            (None, None)
+        };
+
+        ts_data.extend(param_ts_data.into_iter().flatten());
```

Comment:
I think you can move this up into the above `if let Some(v1_parameters) ...` block to avoid wrapping in an `Option`?